benchmarks/asv_compare.conf.json.tpl

```diff
@@ -78,7 +78,9 @@
 
     "build_command" : [
         "python setup.py build {numpy_build_options}",
-        "PIP_NO_BUILD_ISOLATION=false python -mpip wheel --no-deps --no-index -w {build_cache_dir} {build_dir}"
+        // pip ignores '--global-option' when pep517 is enabled, we also enabling pip verbose to
+        // be reached from asv `--verbose` so we can verify the build options.
+        "PIP_NO_BUILD_ISOLATION=false python {build_dir}/benchmarks/asv_pip_nopep517.py -v {numpy_global_options} --no-deps --no-index -w {build_cache_dir} {build_dir}"
```

Comment:
This should be broken into two steps: one to check that `--no-use-pep517` is supported, and one to build the wheel. It might be easier to avoid the pip build altogether by adding a `bdist_wheel` build step
```
rm -rf dist
python setup.py build {numpy_build_options} bdist_wheel
python -mpip install dist/*.whl
```

Reply:
> This should be broken into two steps: one to check that --no-use-pep517 is supported, and one to build the wheel

already handle it through a separate python script `asv_pip_nopep517.py`, we're not free to use POSIX commands since we're targeting non-unix platforms too. e.g. windows

> It might be easier to avoid the pip build altogether by adding a bdist_wheel build step

I tried to use `bdist_wheel` but ASV wouldn't able to fetch the wheel file and cache it.
also, I tried to only use the `pip` command only but idk why `ccache` doesn't work properly

here is the default values of build commands :
```Json
"install_command":
    ["python -mpip install {wheel_file}"],
"uninstall_command":
    ["return-code=any python -mpip uninstall -y {project}"],
"build_command":
    ["python setup.py build",
     "PIP_NO_BUILD_ISOLATION=false python -mpip wheel --no-deps --no-index -w {build_cache_dir} {build_dir}"]
```